utils.go

```diff
@@ -192,3 +192,24 @@ func redactSignature(origAuth string) string {
 	// Strip out 256-bit signature from: Signature=<256-bit signature>
 	return regSign.ReplaceAllString(newAuth, "Signature=**REDACTED**")
 }
+
+// Get default location returns the location based on the input
+// URL `u`, if none of the standard URL matches then if regionOverride
+// is given location gets defaulted to that instead.
+//
+// If no other cases match then the location is set to `us-east-1`
+// as a last resort.
+func getDefaultLocation(u url.URL, regionOverride string) (location string) {
```

Comment:
If the user tries to override AmazonChinaEndpoint region then better not auto correct but let it fail naturally as the server will return error. This will make this function simpler as well.